osu.Game/Storyboards/Commands/StoryboardCommand.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Transforms;
+using osu.Game.Storyboards.Drawables;
+
+namespace osu.Game.Storyboards.Commands
+{
+    public abstract class StoryboardCommand<T> : IStoryboardCommand, IComparable<StoryboardCommand<T>>
+    {
+        public double StartTime { get; }
+        public double EndTime { get; }
+
+        public T StartValue { get; }
+        public T EndValue { get; }
+        public Easing Easing { get; }
+
+        public double Duration => EndTime - StartTime;
+
+        protected StoryboardCommand(Easing easing, double startTime, double endTime, T startValue, T endValue)
+        {
+            if (endTime < startTime)
+                endTime = startTime;
+
+            StartTime = startTime;
+            StartValue = startValue;
+            EndTime = endTime;
+            EndValue = endValue;
+            Easing = easing;
+        }
+
+        public abstract string PropertyName { get; }
+
+        public abstract void ApplyInitialValue<TDrawable>(TDrawable d)
+            where TDrawable : Drawable, IFlippable, IVectorScalable;
+
+        public abstract TransformSequence<TDrawable> ApplyTransforms<TDrawable>(TDrawable d)
+            where TDrawable : Drawable, IFlippable, IVectorScalable;
```

Comment:
Rather than have generics here I'd prefer what https://github.com/ppy/osu/pull/27482 did and have `IDrawableStoryboardElement : IFlippable, IVectorScalable` if possible. I just don't really see that much point here.

(I did initially say in that pull that that felt arbitrary but I guess this explains it better than that diff ever did.)

Reply:
It's not possible to use any interface as a generic type argument for `TransformSequence` because it's constrained to `Drawable` classes, that's why I have the generic parameter here constrained to `Drawable`s.

We can change the `TransformSequence` constraint to `IDrawable`, but I'm not sure if the effort is worth it, so I bailed out and added a *pretty local* generic here.